src/groupNormL2.jl

```diff
@@ -0,0 +1,81 @@
+# Group L2 norm (times a constant)
+
+export GroupNormL2
+
+"""
+**``L_2`` Group - norm**
+    GroupNormL2(λ = 1, g = 1, idx = [1, end])
+Returns the function
+```math
+f(x) =  \\sum\\_{i} \\lambda\\_{i}||x\\_{[i]}||\\_2)^{1/2}
+```
+for groups `x\\_{[i]}` a nonnegative weights `λ\\_i`.
+  Defaults to NormL2() in ProximalOperators if only 1 group is defined.
+"""
+struct GroupNormL2{
+  R <: Real,
+  RR <: AbstractVector{R},
+  I
+  } <: ProximableFunction
+  lambda::RR
+  idx::I
+
+  function GroupNormL2{R, RR,I}(
+    lambda::RR,
+    idx::I
+    ) where {R <: Real, RR <: AbstractVector{R}, I}
+    if any(lambda .< 0)
+      error("weights λ must be nonnegative")
+    elseif length(lambda) != length(idx)
+      error("number of weights and groups must be the same")
+    else
+      new{R, RR, I}(lambda, idx)
+    end
+  end
+end
+
+GroupNormL2(
+  lambda::RR = [1.],
+  idx::I = [Int[]],
+) where {R <: Real, RR <: AbstractVector{R}, I} =
+GroupNormL2{R, RR, I}(lambda, idx)
+
+function (f::GroupNormL2)(x::AbstractArray{T}) where {T <: Real}
+  sum_c = T(0)
+  if length(f.idx) == 1
+    return f.lambda * sqrt(sum(x.^2))
+  else
+    for i = 1:length(f.idx)
+      sum_c += f.lambda[i]*sqrt(sum(x[f.idx[i]].^2))
+    end
+    return sum_c
+  end
+end
+
+function prox!(
+  y::AbstractArray{T},
+  f::GroupNormL2{T, R, I},
```

Comment:
Could you use either `{T, R, I}` or `{R, RR, I}` everywhere?

Reply:
Sure